code/game/objects/structures/flora/generic_flora.dm

```diff
@@ -3,13 +3,178 @@
+				shake_animation()
+				if(do_after(user, 50))
+					timber(user)
+			else
+				to_chat(user, SPAN_WARNING("Try turning \the [C] on first!"))
+				cutting = FALSE
+				return
+		else if(do_after(user, I.w_class * 5)) //Small windup
+			user.do_attack_animation(src)
+			shake_animation()
+			playsound(get_turf(src), 'sound/effects/woodcutting.ogg', 50, 1)
+			chop_health -= I.force
+			update_desc()
+			if(prob(20))
+				user.visible_message(SPAN_NOTICE("\The [user] chops at \the [src]."), SPAN_NOTICE("You chop at \the [src]."), SPAN_NOTICE("You hear someone chopping wood."))
+			if(prob(I.force/3))
+				var/list/valid_turfs = list()
+				for(var/turf/T in circle_range(src, 1))
+					if(T.contains_dense_objects())
+						continue
+					valid_turfs += T
+				if(!valid_turfs.len)
+					valid_turfs += get_turf(src)
+				var/obj/item/stack/material/wood/branch/B = new(pick(valid_turfs))
+				B.amount = 1
+				visible_message(SPAN_NOTICE("\The [B] falls from \the [src]."))
+
+			if(chop_health <= 0)
+				timber(user)
+		cutting = FALSE
+		return
+	..()
+
+/obj/structure/flora/tree/proc/timber(mob/user)
+	var/turf/fall_loc //Where we will fall
+
+	//We prefer to fall directly away from the user if possible
+	var/turf/pref_loc = get_cardinal_step_away(src, user)
+	if(!pref_loc.contains_dense_objects(TRUE))
+		fall_loc = pref_loc
+
+	//Otherwise, we fall elsewhere
+	else
+		var/list/fall_spots = list()
+		for(var/turf/T in orange(1, src))
+			if(locate(user) in T.contents) //Won't fall on woodcutter
+				continue
+			if(T.contains_dense_objects(TRUE)) //Let's avoid dense objects but not mobs
+				continue
+			fall_spots += T
+
+		if(!fall_spots.len)
+			fall_loc = get_turf(src)
+		else
+			fall_loc = pick(fall_spots)
+	playsound(get_turf(src), 'sound/species/diona/gestalt_grow.ogg', 50)
+	for(var/atom/A in fall_loc.contents)
+		if(is_type_in_list(A, fall_forbid))
+			continue
+		if(isliving(A))
+			var/mob/living/L = A
+			visible_message(SPAN_WARNING("\The [src] crushes \the [L] under its weight!"))
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				var/obj/item/organ/external/affected = pick(H.organs)
+				affected.take_damage(fall_force)
+			else
+				if(L.mob_size <= fall_force / 5)
+					L.gib()
+				else
+					L.health -= fall_force
+					if(L.health <= 0)
+						L.gib()
+
+		if(isobj(A) && !istype(A, /obj/item/stack) && A != src)
+			var/obj/O = A
+			if(prob(fall_force * 0.75))
+				visible_message(SPAN_WARNING("\The [src] crushes \the [O] under its weight!"))
+				qdel(A)
+	new /obj/structure/flora/stump/log(fall_loc)
+	var/obj/item/stack/material/wood/branch/B = new(fall_loc)
+	B.amount = rand(5, 8)
+	new stumptype(get_turf(src))
+	qdel(src)
+
+/obj/structure/flora/stump
+	name = "stump"
+	desc = "Nature's chair."
+	icon = 'icons/obj/wood.dmi'
+	icon_state = "tree_stump"
+	density = FALSE
+	anchored = TRUE
+
+/obj/structure/flora/stump/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/shovel))
```

Comment:
Should we have an `can_dig()` / `is_shovel()` proc for this + below? Unsure if it will matter.